tests/models/test_dag.py

```diff
@@ -827,6 +827,15 @@ def test_following_schedule_datetime_timezone(self):
         when = dag.following_schedule(start)
         assert when.isoformat() == "2018-03-25T03:00:00+00:00"
 
+    def test_create_dagrun_when_schedule_is_none_and_empty_start_date(self):
+        # Check that we don't get an AttributeError 'start_date' for self.start_date when schedule is empty
+        dag = DAG("dag_without_start_date")
+        dag.add_task(BaseOperator(task_id="task_without_start_date"))
+        dagrun = dag.create_dagrun(
+            state=State.RUNNING, run_type=DagRunType.SCHEDULED, execution_date=DEFAULT_DATE
```

Comment:
How can `DagRunType` be scheduled ? With scheduled runs we must have `start_date`

Reply:
> How can DagRunType be scheduled ? With scheduled runs we must have start_date

IMHO it should be ok if catchup is `False` 🤔

Reply:
The scheduler creates runs with type scheduled. The fix in this PR is for manual runs that are created by the user.

catchup parameter has no affect on manual runs

Reply:
Yes, but I wonder if we can do the same thing for dags with `schedule!=None` if the catchup is set to `False`, wdyt?

Reply:
I dont think so because the first run depends on the start_date. If you set it in the past and interval passed when dag is activated it will create a run. If you set it for a future date then interval is not completed thus no run will be created when dag is set to active.

Reply:
yes, but we can consider that this condition is always met when the user does not provide a start date.